src/screens/PostsDisplayScreen/PostCard/PostCard.tsx

```diff
@@ -0,0 +1,374 @@
+import {useDispatch, useSelector} from "react-redux";
+import React, {useEffect, useState} from "react";
+import axios from "axios";
+import {URI} from "../../../URI";
+import {TouchableOpacity, View, Image, TouchableWithoutFeedback, Text, Modal, StyleSheet} from "react-native";
+import {addLikes, removeLikes} from "../../../redux/actions/postActions";
+import PostDetailsCard from "./PostDetailsCard";
+import getTimeDuration from "./timeGen";
+import {useNavigation} from "@react-navigation/native";
+import {Post, User} from "../../../redux/reducers/User";
+import {getPosts} from "../../../redux/actions/getPosts";
+
+type Props = {
+    item: Post;
+    isReply?: boolean | null;
+    postId?: string | null;
+    replies?: boolean | null;
+};
+
+export const PostCard = ({item, isReply, postId, replies}: Props) => {
+    const navigation = useNavigation();
+    const {user, token, users} = useSelector((state: any) => state.user);
+    const {posts} = useSelector((state: any) => state.post);
+    const [openModal, setOpenModal] = useState(false);
+    const dispatch = useDispatch();
+    const [userInfo, setUserInfo] = useState({
+        name: '',
+        avatar: {
+            url: 'https://res.cloudinary.com/dshp9jnuy/image/upload/v1665822253/avatars/nrxsg8sd9iy10bbsoenn.png',
+        },
+    });
+
+    const time = item?.createdAt;
```

Comment:
We're working with `null`s in a wrong way here. Let's think about it high level. We have a component which renders a post. Can it work if it doesn't have the post data? No. The post data is essential for this component. If we don't have data, we have nothing to show. If we have nothing to show, we should not render the component. 

We can do an `if` in the parent component like this:
```tsx
<>
    {item ? <PostCard item={item}> : null}
</>
```
This way we can be confident inside our component that data is present. We don't need to do the checks on line 34 and optional chaining on line 53. 

Using optional chaining and definite assignment assertions can be useful sometimes, but it should not be used instead of accurate data modelling and runtime checks (see more https://medium.com/ngconf/the-hidden-danger-of-optional-chaining-552a6a45b1d1)